### PR TITLE
fix(website): remove empty index pages

### PR DESCRIPTION
**Pull request by zchsh**
👀 [Preview](https://packer-git-zsrm-empty-pages-hashicorp.vercel.app)
🎟️ [Asana task](https://app.asana.com/0/1100423001970639/1202059456316732/f)

This PR removes empty overview pages from the website. The intent behind this removal is to improve user experience - it can be a strange experience to click on page only to find it empty.

It's likely these pages are a holdover from past iterations of the website where Overview pages were required.

### CODE REVIEW

File: website/data/docs-nav-data.json [hunk 1]
```diff
@@ -154,10 +154,6 @@
               {
                 "title": "Contextual Functions",
                 "routes": [
-                  {
```
Comment (zchsh) [inline website/data/docs-nav-data.json:157]: These sidebar items currently lead to empty pages. We want to remove the empty pages, so we need to remove these sidebar items as well.

**Comment by zchsh** [comment]
@BRKalow Like https://github.com/hashicorp/vault/pull/14972 but for some stray Packer pages 👍 Also less complex since there's no need to remove a whole `[[...page]].tsx` file here.

**Review by BRKalow — APPROVED** [review]
🧹

**Review by nywilken — APPROVED** [review]
LGTM

**Pull request merged**